operator/v1/types_network.go

```diff
@@ -100,12 +100,90 @@ type DefaultNetworkDefinition struct {
 	KuryrConfig *KuryrConfig `json:"kuryrConfig,omitempty"`
 }
 
+// SimpleMacvlanConfig contains configurations for macvlan interface.
+type SimpleMacvlanConfig struct {
+	// master is the host interface to create the macvlan interface from.
+	// If not specified, it will be default route interface
+	// +optional
+	Master string `json:"master,omitempty"`
+
+	// IPAMConfig configures IPAM module will be used for IP Address Management (IPAM).
+	// +optional
+	IPAMConfig *IPAMConfig `json:"ipamConfig,omitempty"`
+
+	// mode is the macvlan mode: bridge, private, vepa, passthru. The default is bridge
+	// +optional
+	Mode MacvlanMode `json:"mode,omitempty"`
+
+	// mtu is the mtu to use for the macvlan interface. if unset, host's
+	// kernel will select the value.
+	// +optional
+	MTU uint32 `json:"mtu,omitempty"`
+}
+
+// StaticIPAMAddresses provides IP address and Gateway for static IPAM addresses
+type StaticIPAMAddresses struct {
+	// Address is the IP address in CIDR format
+	// +optional
+	Address string `json:"address"`
+	// Gateway is IP inside of subnet to designate as the gateway
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+}
+
+// StaticIPAMRoutes provides Destination/Gateway pairs for static IPAM routes
+type StaticIPAMRoutes struct {
+	// Destination points the IP route destination
+	Destination string `json:"destination"`
+	// Gateway is the route's next-hop IP address
+	// If unset, a default gateway is assumed (as determined by the CNI plugin).
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+}
+
+// StaticIPAMDNS provides DNS related information for static IPAM
+type StaticIPAMDNS struct {
```

Comment:
I know this is in the CNI type, but does anything actually use this? Does CRI-O actually update `/etc/resolv.conf` if this is set? AFAICT, it does not. So we should probably just remove DNS? Or is there a use for it I'm missing?

Reply:
Actually DNS may not be used for now (blocked to input by operator), but for consistency between API and IPAM spec, we should keep it in API, I suppose.